src/ext/framework/mod.rs

```diff
@@ -258,6 +258,50 @@ impl Framework {
         self
     }
 
+    /// Aliases a command, allowing it to be used under a different name.
+    ///
+    /// Note that the default group is named "Ungrouped".
+    pub fn alias<S>(mut self, group: &str, to: &str, from: &str) -> Self {
+        {
+            let group = self.groups.entry(group.to_owned())
+                .or_insert_with(|| Arc::new(CommandGroup::default()));
+
+            if let Some(ref mut group) = Arc::get_mut(group) {
+                if let Some(ref prefix) = group.prefix {
+                    group.commands.insert(format!("{} {}", prefix, from.to_owned()), CommandOrAlias::Alias(format!("{} {}", prefix, to.to_owned())));
+                } else {
+                    group.commands.insert(from.to_owned(), CommandOrAlias::Alias(to.to_owned()));
+                }
+            }
+        }
+
+        self
+    }
+
+    /// Shorthand way to add multiple aliases.
+    ///
+    /// Note that the default group is named "Ungrouped".
+    pub fn aliases<S>(mut self, group: &str, from: &str, to: Vec<&str>) -> Self {
```

Comment:
`<S>` is no longer needed; users would also experience an error about not being able to infer enough information about the trait bound on use. See the `alias` method as well